src/globus_cli/commands/gcs/endpoint/role/list.py

```diff
@@ -0,0 +1,42 @@
+import uuid
+
+import click
+from globus_sdk.paging import Paginator
+
+from globus_cli.commands.gcs.endpoint.role._common import role_fields
+from globus_cli.login_manager import LoginManager
+from globus_cli.parsing import command, endpoint_id_arg
+from globus_cli.services.transfer import iterable_response_to_dict
+from globus_cli.termio import display
+from globus_cli.utils import PagingWrapper
+
+
+@command(
+    "show",
+)
+@endpoint_id_arg
+@click.option(
+    "--all-roles", is_flag=True, help="Show all roles, not just yours.", default=False
+)
+@LoginManager.requires_login("transfer")
+def list_command(
+    login_manager: LoginManager,
+    *,
+    endpoint_id: uuid.UUID,
+    all_roles: bool,
+) -> None:
+    """List all roles on a GCS Endpoint associated with you."""
+    gcs_client = login_manager.get_gcs_client(endpoint_id=endpoint_id)
+    auth_client = login_manager.get_auth_client()
+
+    paginator = Paginator.wrap(gcs_client.get_role_list)
+    if all_roles:
+        res = PagingWrapper(paginator(include="all_roles").items())
+    else:
+        res = PagingWrapper(paginator().items())
```

Comment:
I want to note and acknowledge -- but explicitly not try to address here -- that it would be nice to have the `json_converter` behavior built into `PagingWrapper` more deeply. These pieces were added incrementally over time, and could do with a bit of a clean redesign.
(Also setting aside the idea of changing how CLI pagination should work: I mean that even if we're keeping exactly the same behavior, we could probably have a tidier implementation.)

I'm glad that the tools we have continue to work to put together new commands, but there's nothing like seeing it in code review to drive home that these pieces don't quite click together the way I'd like.

Reply:
Yeah, I considered peeking under the hood to iterate on this design (no pun intended) but had already let a bit of scope creep into this change with the principal type resolution & didn't want to add more.